### PR TITLE
KAFKA-6145: KIP-441: Add test scenarios to ensure rebalance convergence

### PR DESCRIPTION
**Pull request by vvcephei**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/assignment/AssignmentTestUtils.java [hunk 1]
```diff
@@ -63,24 +63,6 @@
...
-            for (int j = 0; j < 4; ++j) {
-                builder.append(n);
-            }
-            builder.append('-');
-        }
-        for (int i = 0; i < 12; ++i) {
-            builder.append(n);
-        }
-        return UUID.fromString(builder.toString());
+        return new UUID(0, n);
```
Comment (ableegoldman) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/assignment/AssignmentTestUtils.java:66]: Oh wow, I can't believe I didn't notice this constructor existed 🤦‍♀️

Comment (vvcephei) [reply streams/src/test/java/org/apache/kafka/streams/processor/internals/assignment/AssignmentTestUtils.java:66]: I missed it, too during reviews :shrug:

**Comment by vvcephei** [comment]
Note to reviewers, the new tests here actually highlighted some issues with the current implementation. We'll fix the implementation in a follow-on, and keep this PR only to adding the tests, so the failling ones are marked as "ignored" for now.

See https://github.com/apache/kafka/pull/8497/ for the pending fix.

**Review by ableegoldman — APPROVED** [review]
LGTM, thanks for the tests! Fix coming up next

**Comment by vvcephei** [comment]
java 11 test failures were unrelated:
    kafka.api.ConsumerBounceTest.testRollingBrokerRestartsWithSmallerMaxGroupSizeConfigDisruptsBigGroup
    kafka.api.ConsumerBounceTest.testRollingBrokerRestartsWithSmallerMaxGroupSizeConfigDisruptsBigGroup
    org.apache.kafka.streams.integration.EOSUncleanShutdownIntegrationTest.shouldWorkWithUncleanShutdownWipeOutStateStore[exactly_once_beta]

**Pull request merged**